### PR TITLE
add "local" remotes

### PR DESCRIPTION
**Pull request by alcortesm**
By using the "dir://" scheme, remotes from a local ".git" repository are now allowed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -87,6 +87,29 @@ Date:   2015-12-11 17:57:10 +0100 +0100
 ...
 ```
 
+Bare and non-bare local repositories are also supported using the `dir://`
```
Comment (mcuadros) [inline README.md]: what about using `local://` as scheme for the local remotes?

File: blame_test.go [hunk 2]
```diff
@@ -25,13 +25,13 @@ func (s *BlameCommon) SetUpSuite(c *C) {
 		d, err := os.Open(fixRepo.packfile)
 		c.Assert(err, IsNil)
 
-		r := packfile.NewReader(d)
+		decoder := packfile.NewDecoder(d)
```
Comment (mcuadros) [inline blame_test.go]: I prefer keep short names to short scope variables

File: clients/common.go [hunk 3]
```diff
@@ -16,6 +16,7 @@ import (
 	"net/url"
 
 	"gopkg.in/src-d/go-git.v3/clients/common"
+	"gopkg.in/src-d/go-git.v3/clients/file"
```
Comment (mcuadros) [inline clients/common.go]: really we  must to choose, "local", "dir" or "file" as client name.

File: clients/common_test.go [hunk 4]
```diff
@@ -3,18 +3,36 @@ package clients
...
 func Test(t *testing.T) { TestingT(t) }
 
-type SuiteCommon struct{}
+type SuiteCommon struct {
+	dirFixturePath string
+}
 
 var _ = Suite(&SuiteCommon{})
 
+const fixtureTGZ = "../formats/gitdir/fixtures/spinnaker-gc.tgz"
```
Comment (mcuadros) [inline clients/common_test.go]: gitdir, call the dir as the name of the client

File: README.md (hunk 1 above)
Comment (alcortesm) [inline README.md]: `local://` will be just perfect.

File: blame_test.go (hunk 2 above)
Comment (alcortesm) [inline blame_test.go]: OK, will do.

File: clients/common.go (hunk 3 above)
Comment (alcortesm) [inline clients/common.go]: ok, local would be perfect if we are going to use `local://` as the scheme.

Comment (alcortesm) [inline clients/common.go]: Altough, it is not really a client as there is no server, but that is a different problem we must solve :).

File: clients/common_test.go (hunk 4 above)
Comment (alcortesm) [inline clients/common_test.go]: OK, will do.

**Comment by alcortesm** [comment]
I'll remove go 1.5 from the CI, so the travis don't complain anymore about my builds.

**Comment by sona-tar** [comment]
file:// scheme is more common than the local:// scheme.

https://en.m.wikipedia.org/wiki/File_URI_scheme
https://docs.oracle.com/javase/8/docs/api/java/io/File.html

And we can use file:// scheme as link on Windows.

**Comment by alcortesm** [comment]
You are right @sona-tar , thanks for the advice!

**Comment by sona-tar** [comment]
@alcortesm Thanks!! I'm looking forward to merging file scheme to master.

**Comment by alcortesm** [comment]
As per instructed by @mcuadros I'm closing this PR.

All the features in this PR will be included in a future PR along with some new features a quite an important refactor.

**Pull request closed without merging**